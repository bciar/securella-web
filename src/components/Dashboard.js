import React, { Component } from 'react';
import $ from 'jquery';
import ApiURL from '../apiClient';
import { RaisedButton, Paper } from 'material-ui';

import { ValidatorForm } from 'react-form-validator-core';
import { TextValidator} from 'react-material-ui-form-validator';

export class Dashboard extends Component {

  constructor(props) {
      super(props);
      this.state = {
          alarmData: {
            status: "closed",
            latitude: -33.931498,
            longitude: 18.434765
          },
          formData: {
              latitude: '-33.931498',
              longitude: '18.434765'
          },
          submitted: false,
          errors: false
      };
  }

  componentWillMount() {
    if (!this.props.user.loggedIn) {
      this.props.history.push('/login');
    }

    return $.ajax ({
      url: ApiURL("user/alarm_status"),
      type: "GET",
      dataType: "json",
      contentType: "application/json",
      success: function(data) { this.setState({alarmData: data}); console.log(data); }.bind(this)
    });
  }

  handleChange = (event) => {
      const { alarmData } = this.state;
      alarmData[event.target.name] = event.target.value;
      this.setState({ alarmData });
  }

  newAlarm = (history) => {
    return $.ajax ({
      url: ApiURL("user/new_alarm"),
      type: "POST",
      data: JSON.stringify({
        "alarm": {
        "latitude": document.getElementById('latitude').value,
        "longitude": document.getElementById('longitude').value}
      }),
      dataType: "json",
      contentType: "application/json",
      success: function(data) { this.setState({alarmData: data}); console.log(data); }.bind(this)
    });
  }

  updateAlarm = (history) => {
    return $.ajax ({
      url: ApiURL("user/new_alarm"),
      type: "PUT",
      data: JSON.stringify({
        "alarm": {
        "latitude": document.getElementById('latitude').value,
        "longitude": document.getElementById('longitude').value}
      }),
      dataType: "json",
      contentType: "application/json",
      success: function(data) { this.setState({alarmData: data}); console.log(data); }.bind(this)
    });
  }

  closeAlarm = () => {
    return $.ajax ({
      url: ApiURL("user/close_alarm"),
      type: "POST",
      dataType: "json",
      contentType: "application/json",
      success: function(data) { this.setState({alarmData: data}); console.log(data); }.bind(this)
    });
  }

  updateStatus = () => {
    return $.ajax ({
      url: ApiURL("user/alarm_status"),
      type: "GET",
      dataType: "json",
      contentType: "application/json",
      success: function(data) { this.setState({alarmData: data}); console.log(data); }.bind(this)
    });
  }

  render() {
    const { alarmData, submitted } = this.state;
    if (alarmData.status === "open") {
      return (
        <div className="container">
        <Paper className="common-paper" zDepth={2}>
          <div>
            <h3>Alarm: {this.state.alarmData.status}</h3>
            <ValidatorForm
                ref="form"
                name="alarmform"
                onSubmit={ () => this.newAlarm(this.props.history) } >
                <TextValidator
                    floatingLabelText="Latiutude"
                    onChange={this.handleChange}
                    name="latitude"
                    id="latitude"
                    value={alarmData.latitude}
                    validators={['required']}
                    errorMessages={['this field is required']}
                />
                <br />
                <TextValidator
                    floatingLabelText="Longitude"
                    onChange={this.handleChange}
                    name="longitude"
                    id="longitude"
                    validators={['required']}
                    errorMessages={['this field is required']}
                    value={alarmData.longitude}
                />
                <br />
                <RaisedButton
                    type="submit"
                    name="submitbutton"
                    label={
                        (submitted && 'Processing...')
                        || (!submitted && 'Update Alarm')
                    }
                    disabled={submitted}
                />
            </ValidatorForm>
          </div>
          <div className="common-paper-button-wrapper">
            <RaisedButton label="Update View" onClick={ () => this.updateStatus() }/>
            <RaisedButton label="Close Alarm" onClick={ () => this.closeAlarm() }/>
          </div>
        </Paper>
        </div>
      );
    }
    else {
      return (
        <div className="container">
        <Paper className="common-paper" zDepth={2}>
          <div>
            <h3>Alarm: {this.state.alarmData.status}</h3>
            <ValidatorForm
                ref="form"
                name="alarmform"
                onSubmit={ () => this.newAlarm(this.props.history) } >
                <TextValidator
                    floatingLabelText="Latiutude"
                    onChange={this.handleChange}
                    name="latitude"
                    id="latitude"
                    value={alarmData.latitude}
                    validators={['required']}
                    errorMessages={['this field is required']}
                />
                <br />
                <TextValidator
                    floatingLabelText="Longitude"
                    onChange={this.handleChange}
                    name="longitude"
                    id="longitude"
                    validators={['required']}
                    errorMessages={['this field is required']}
                    value={alarmData.longitude}
                />
                <br />
                <RaisedButton
                    type="submit"
                    name="submitbutton"
                    label={
                        (submitted && 'Processing...')
                        || (!submitted && 'New Alarm')
                    }
                    disabled={submitted}
                />
            </ValidatorForm>
          </div>
          <div className="common-paper-button-wrapper">
            <RaisedButton label="Update View" onClick={ () => this.updateStatus() }/>
          </div>
        </Paper>
        </div>
      );
    }
  }
}
