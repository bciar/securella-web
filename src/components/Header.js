import React from 'react';
import { Link } from 'react-router-dom';
import AppBar from 'material-ui/AppBar';
import {ToolbarGroup, ToolbarTitle} from 'material-ui/Toolbar';
import FlatButton from 'material-ui/FlatButton';

const styles = {
  appbar: {
    background: '#393'
  }
}

const PublicNavLinks = () => (
  <ToolbarGroup>
    <ToolbarTitle />
    <FlatButton label="Home" containerElement={<Link to='/'>Home</Link>}/>
    <FlatButton label="Login" containerElement={<Link to='/login'>login</Link>}/>
    <FlatButton label="Register" containerElement={<Link to='/signup'>signup</Link>}/>
  </ToolbarGroup>
);

const MyNavLinks = () => (
  <ToolbarGroup>
    <ToolbarTitle />
    <FlatButton label="Dashboard" containerElement={<Link to='/dashboard'>Dashboard</Link>}/>
    <FlatButton label="My Account" containerElement={<Link to='/account'>account</Link>}/>
    <FlatButton label="Logout" containerElement={<Link to='/logout'>logout</Link>}/>
  </ToolbarGroup>
);

const EmptyNavLinks = () => (
  <ToolbarGroup>
    <ToolbarTitle />
  </ToolbarGroup>
);

export const Head = ({ app, user }) => {
    if (app.loaded) {
        if (user.loggedIn) {
            return ( 
              <AppBar title="SEON" 
                      style={styles.appbar} 
                      showMenuIconButton={false} 
                      iconElementRight={<MyNavLinks />} 
              /> 
            );
        }
        else {
            return ( 
              <AppBar title="SEON" 
                      style={styles.appbar} 
                      showMenuIconButton={false} 
                      iconElementRight={<PublicNavLinks />} 
              /> 
            );
        }
    }
    else {
        return ( 
            <AppBar title="SEON" 
                    style={styles.appbar} 
                    showMenuIconButton={false} 
                    iconElementRight={<EmptyNavLinks />} 
            /> 
        );
    }
}