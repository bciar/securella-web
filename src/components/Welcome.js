import React, { Component } from 'react';
import { Link, Redirect } from 'react-router-dom';
import { Paper} from 'material-ui';

export class Welcome extends Component {
  
    constructor(props) {
        super(props);

        this.state = {
          username: ''
        };
    }

    renderApp() {
        if (this.props.user.loggedIn) { 
          return <Redirect to='/dashboard' />; 
        }
        else {
            return (
                <div className="container">
                    <Paper className="common-paper">
                        <h1>Welcome to Securella</h1>
                        <p>
                          <Link to='/signup'>Register</Link> if you are new here.
                          <Link to='/login'>Login</Link> if you already have an account.
                        </p>
                    </Paper>
                </div>
            );
        }
    }

    render() {
        return this.props.app.loaded ? this.renderApp() : ('Loading...')
    }
}
