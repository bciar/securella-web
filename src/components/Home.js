import React, { Component } from 'react';
import {Switch, Route}      from 'react-router-dom';

import {Welcome}    from './Welcome.js';
import {Dashboard}  from './Dashboard.js';

import {SignUp}       from './user/Sign_up.js';
import {VerifyEmail}  from './user/VerifyEmail.js';
import {ConfirmEmail} from './user/ConfirmEmail.js';
import {SignIn}       from './user/Sign_in.js';
import {SignOut}      from './user/Sign_out.js';

import {RequestPasswordReset}   from './user/RequestPasswordReset.js';
import {RequestPasswordSuccess} from './user/RequestPasswordSuccess.js';
import {ResetPassword}          from './user/ResetPassword.js';
import {ResetPasswordSuccess}   from './user/ResetPasswordSuccess.js';

import {Account}       from './user/Account.js';
import {UpdateProfile} from './user/UpdateProfile.js';
import {UserImage}     from './user/UserImage.js';

export class Home extends Component {
  render() {
    return (
      <div>
        <Switch>
          <Route exact path='/'               render={(props) => ( <Welcome   {...props} user={this.props.user} app={this.props.app} /> )} />
          <Route exact path='/dashboard'      render={(props) => ( <Dashboard {...props} user={this.props.user} app={this.props.app} /> )} />

          <Route exact path='/signup'         render={(props) => ( <SignUp       {...props} user={this.props.user} app={this.props.app} /> )} />
          <Route exact path='/confirm'        render={(props) => ( <ConfirmEmail {...props} user={this.props.user} app={this.props.app} /> )} />
          <Route exact path='/verify'         render={(props) => ( <VerifyEmail  {...props} user={this.props.user} app={this.props.app} /> )} />

          <Route exact path='/login'          render={(props) => ( <SignIn  {...props} user={this.props.user} app={this.props.app} /> )} />
          <Route exact path='/logout'         render={(props) => ( <SignOut {...props} user={this.props.user} app={this.props.app} /> )} />

          <Route exact path='/request_password_reset'   render={(props) => ( <RequestPasswordReset   {...props} user={this.props.user} app={this.props.app} /> )} />
          <Route exact path='/request_password_success' render={(props) => ( <RequestPasswordSuccess {...props} user={this.props.user} app={this.props.app} /> )} />
          <Route exact path='/reset_password'           render={(props) => ( <ResetPassword          {...props} user={this.props.user} app={this.props.app} /> )} />
          <Route exact path='/reset_password_success'   render={(props) => ( <ResetPasswordSuccess   {...props} user={this.props.user} app={this.props.app} /> )} />

          <Route exact path='/account'        render={(props) => ( <Account        {...props} user={this.props.user} app={this.props.app} /> )} />
          <Route exact path='/update_profile' render={(props) => ( <UpdateProfile  {...props} user={this.props.user} app={this.props.app} /> )} />
          <Route exact path='/update_image'   render={(props) => ( <UserImage      {...props} user={this.props.user} app={this.props.app} /> )} />
        </Switch>
      </div>
    );
  }
}
