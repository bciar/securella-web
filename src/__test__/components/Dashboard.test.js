import React from 'react';
import { shallow } from 'enzyme';
import { Dashboard } from '../../components/Dashboard';

let test_alarm_data = {
    status: "closed",
    latitude: 37.0212334, 
    longitude: -81.234235
};

let user, history, wrapper;
beforeEach(() => {
    user = jest.fn();
    history = { push: jest.fn() };
    wrapper = shallow(<Dashboard user={user} history={history} />);
});

test('Dashboard component should be rendered', () => {
    wrapper.setState({ alarmData: test_alarm_data });
    expect(wrapper).toMatchSnapshot();
});

test('alarm data should be set when the TextValidator is changed', () => {
    wrapper.find('TextValidator').at(0).simulate('change', {
        target: { name: 'latitude', value: test_alarm_data.latitude }
    });
    wrapper.find('TextValidator').at(1).simulate('change', {
        target: { name: 'longitude', value: test_alarm_data.longitude }
    });
    expect(wrapper.state().alarmData).toEqual(test_alarm_data);
});