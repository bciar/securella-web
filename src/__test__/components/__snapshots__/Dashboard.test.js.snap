// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component should be rendered 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dashboard
    history={
        Object {
            "push": [Function],
          }
    }
    user={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Paper
        circle={false}
        className="common-paper"
        rounded={true}
        transitionEnabled={true}
        zDepth={2}
>
        <div>
                <h3>
                        Alarm: 
                        closed
                </h3>
                <ValidatorForm
                        name="alarmform"
                        onError={[Function]}
                        onSubmit={[Function]}
                >
                        <TextValidator
                                errorMessages={
                                        Array [
                                                "this field is required",
                                              ]
                                }
                                floatingLabelText="Latiutude"
                                id="latitude"
                                name="latitude"
                                onChange={[Function]}
                                validatorListener={[Function]}
                                validators={
                                        Array [
                                                "required",
                                              ]
                                }
                                value={37.0212334}
                        />
                        <br />
                        <TextValidator
                                errorMessages={
                                        Array [
                                                "this field is required",
                                              ]
                                }
                                floatingLabelText="Longitude"
                                id="longitude"
                                name="longitude"
                                onChange={[Function]}
                                validatorListener={[Function]}
                                validators={
                                        Array [
                                                "required",
                                              ]
                                }
                                value={-81.234235}
                        />
                        <br />
                        <RaisedButton
                                disabled={false}
                                fullWidth={false}
                                label="New Alarm"
                                labelPosition="after"
                                name="submitbutton"
                                primary={false}
                                secondary={false}
                                type="submit"
                        />
                </ValidatorForm>
        </div>
        <div
                className="common-paper-button-wrapper"
        >
                <RaisedButton
                        disabled={false}
                        fullWidth={false}
                        label="Update View"
                        labelPosition="after"
                        onClick={[Function]}
                        primary={false}
                        secondary={false}
                />
        </div>
</Paper>,
      "className": "container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div>
            <h3>
                        Alarm: 
                        closed
            </h3>
            <ValidatorForm
                        name="alarmform"
                        onError={[Function]}
                        onSubmit={[Function]}
            >
                        <TextValidator
                                    errorMessages={
                                                Array [
                                                            "this field is required",
                                                          ]
                                    }
                                    floatingLabelText="Latiutude"
                                    id="latitude"
                                    name="latitude"
                                    onChange={[Function]}
                                    validatorListener={[Function]}
                                    validators={
                                                Array [
                                                            "required",
                                                          ]
                                    }
                                    value={37.0212334}
                        />
                        <br />
                        <TextValidator
                                    errorMessages={
                                                Array [
                                                            "this field is required",
                                                          ]
                                    }
                                    floatingLabelText="Longitude"
                                    id="longitude"
                                    name="longitude"
                                    onChange={[Function]}
                                    validatorListener={[Function]}
                                    validators={
                                                Array [
                                                            "required",
                                                          ]
                                    }
                                    value={-81.234235}
                        />
                        <br />
                        <RaisedButton
                                    disabled={false}
                                    fullWidth={false}
                                    label="New Alarm"
                                    labelPosition="after"
                                    name="submitbutton"
                                    primary={false}
                                    secondary={false}
                                    type="submit"
                        />
            </ValidatorForm>
</div>,
          <div
            className="common-paper-button-wrapper"
>
            <RaisedButton
                        disabled={false}
                        fullWidth={false}
                        label="Update View"
                        labelPosition="after"
                        onClick={[Function]}
                        primary={false}
                        secondary={false}
            />
</div>,
        ],
        "circle": false,
        "className": "common-paper",
        "rounded": true,
        "transitionEnabled": true,
        "zDepth": 2,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                Alarm: 
                closed
</h3>,
              <ValidatorForm
                name="alarmform"
                onError={[Function]}
                onSubmit={[Function]}
>
                <TextValidator
                                errorMessages={
                                                Array [
                                                                "this field is required",
                                                              ]
                                }
                                floatingLabelText="Latiutude"
                                id="latitude"
                                name="latitude"
                                onChange={[Function]}
                                validatorListener={[Function]}
                                validators={
                                                Array [
                                                                "required",
                                                              ]
                                }
                                value={37.0212334}
                />
                <br />
                <TextValidator
                                errorMessages={
                                                Array [
                                                                "this field is required",
                                                              ]
                                }
                                floatingLabelText="Longitude"
                                id="longitude"
                                name="longitude"
                                onChange={[Function]}
                                validatorListener={[Function]}
                                validators={
                                                Array [
                                                                "required",
                                                              ]
                                }
                                value={-81.234235}
                />
                <br />
                <RaisedButton
                                disabled={false}
                                fullWidth={false}
                                label="New Alarm"
                                labelPosition="after"
                                name="submitbutton"
                                primary={false}
                                secondary={false}
                                type="submit"
                />
</ValidatorForm>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Alarm: ",
                  "closed",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Alarm: ",
                "closed",
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <TextValidator
                    errorMessages={
                                        Array [
                                                            "this field is required",
                                                          ]
                    }
                    floatingLabelText="Latiutude"
                    id="latitude"
                    name="latitude"
                    onChange={[Function]}
                    validatorListener={[Function]}
                    validators={
                                        Array [
                                                            "required",
                                                          ]
                    }
                    value={37.0212334}
/>,
                  <br />,
                  <TextValidator
                    errorMessages={
                                        Array [
                                                            "this field is required",
                                                          ]
                    }
                    floatingLabelText="Longitude"
                    id="longitude"
                    name="longitude"
                    onChange={[Function]}
                    validatorListener={[Function]}
                    validators={
                                        Array [
                                                            "required",
                                                          ]
                    }
                    value={-81.234235}
/>,
                  <br />,
                  <RaisedButton
                    disabled={false}
                    fullWidth={false}
                    label="New Alarm"
                    labelPosition="after"
                    name="submitbutton"
                    primary={false}
                    secondary={false}
                    type="submit"
/>,
                ],
                "name": "alarmform",
                "onError": [Function],
                "onSubmit": [Function],
              },
              "ref": "form",
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "errorMessages": Array [
                      "this field is required",
                    ],
                    "floatingLabelText": "Latiutude",
                    "id": "latitude",
                    "name": "latitude",
                    "onChange": [Function],
                    "validatorListener": [Function],
                    "validators": Array [
                      "required",
                    ],
                    "value": 37.0212334,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "errorMessages": Array [
                      "this field is required",
                    ],
                    "floatingLabelText": "Longitude",
                    "id": "longitude",
                    "name": "longitude",
                    "onChange": [Function],
                    "validatorListener": [Function],
                    "validators": Array [
                      "required",
                    ],
                    "value": -81.234235,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "disabled": false,
                    "fullWidth": false,
                    "label": "New Alarm",
                    "labelPosition": "after",
                    "name": "submitbutton",
                    "primary": false,
                    "secondary": false,
                    "type": "submit",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <RaisedButton
              disabled={false}
              fullWidth={false}
              label="Update View"
              labelPosition="after"
              onClick={[Function]}
              primary={false}
              secondary={false}
/>,
            "className": "common-paper-button-wrapper",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "disabled": false,
              "fullWidth": false,
              "label": "Update View",
              "labelPosition": "after",
              "onClick": [Function],
              "primary": false,
              "secondary": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Paper
          circle={false}
          className="common-paper"
          rounded={true}
          transitionEnabled={true}
          zDepth={2}
>
          <div>
                    <h3>
                              Alarm: 
                              closed
                    </h3>
                    <ValidatorForm
                              name="alarmform"
                              onError={[Function]}
                              onSubmit={[Function]}
                    >
                              <TextValidator
                                        errorMessages={
                                                  Array [
                                                            "this field is required",
                                                          ]
                                        }
                                        floatingLabelText="Latiutude"
                                        id="latitude"
                                        name="latitude"
                                        onChange={[Function]}
                                        validatorListener={[Function]}
                                        validators={
                                                  Array [
                                                            "required",
                                                          ]
                                        }
                                        value={37.0212334}
                              />
                              <br />
                              <TextValidator
                                        errorMessages={
                                                  Array [
                                                            "this field is required",
                                                          ]
                                        }
                                        floatingLabelText="Longitude"
                                        id="longitude"
                                        name="longitude"
                                        onChange={[Function]}
                                        validatorListener={[Function]}
                                        validators={
                                                  Array [
                                                            "required",
                                                          ]
                                        }
                                        value={-81.234235}
                              />
                              <br />
                              <RaisedButton
                                        disabled={false}
                                        fullWidth={false}
                                        label="New Alarm"
                                        labelPosition="after"
                                        name="submitbutton"
                                        primary={false}
                                        secondary={false}
                                        type="submit"
                              />
                    </ValidatorForm>
          </div>
          <div
                    className="common-paper-button-wrapper"
          >
                    <RaisedButton
                              disabled={false}
                              fullWidth={false}
                              label="Update View"
                              labelPosition="after"
                              onClick={[Function]}
                              primary={false}
                              secondary={false}
                    />
          </div>
</Paper>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div>
              <h3>
                            Alarm: 
                            closed
              </h3>
              <ValidatorForm
                            name="alarmform"
                            onError={[Function]}
                            onSubmit={[Function]}
              >
                            <TextValidator
                                          errorMessages={
                                                        Array [
                                                                      "this field is required",
                                                                    ]
                                          }
                                          floatingLabelText="Latiutude"
                                          id="latitude"
                                          name="latitude"
                                          onChange={[Function]}
                                          validatorListener={[Function]}
                                          validators={
                                                        Array [
                                                                      "required",
                                                                    ]
                                          }
                                          value={37.0212334}
                            />
                            <br />
                            <TextValidator
                                          errorMessages={
                                                        Array [
                                                                      "this field is required",
                                                                    ]
                                          }
                                          floatingLabelText="Longitude"
                                          id="longitude"
                                          name="longitude"
                                          onChange={[Function]}
                                          validatorListener={[Function]}
                                          validators={
                                                        Array [
                                                                      "required",
                                                                    ]
                                          }
                                          value={-81.234235}
                            />
                            <br />
                            <RaisedButton
                                          disabled={false}
                                          fullWidth={false}
                                          label="New Alarm"
                                          labelPosition="after"
                                          name="submitbutton"
                                          primary={false}
                                          secondary={false}
                                          type="submit"
                            />
              </ValidatorForm>
</div>,
            <div
              className="common-paper-button-wrapper"
>
              <RaisedButton
                            disabled={false}
                            fullWidth={false}
                            label="Update View"
                            labelPosition="after"
                            onClick={[Function]}
                            primary={false}
                            secondary={false}
              />
</div>,
          ],
          "circle": false,
          "className": "common-paper",
          "rounded": true,
          "transitionEnabled": true,
          "zDepth": 2,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3>
                  Alarm: 
                  closed
</h3>,
                <ValidatorForm
                  name="alarmform"
                  onError={[Function]}
                  onSubmit={[Function]}
>
                  <TextValidator
                                    errorMessages={
                                                      Array [
                                                                        "this field is required",
                                                                      ]
                                    }
                                    floatingLabelText="Latiutude"
                                    id="latitude"
                                    name="latitude"
                                    onChange={[Function]}
                                    validatorListener={[Function]}
                                    validators={
                                                      Array [
                                                                        "required",
                                                                      ]
                                    }
                                    value={37.0212334}
                  />
                  <br />
                  <TextValidator
                                    errorMessages={
                                                      Array [
                                                                        "this field is required",
                                                                      ]
                                    }
                                    floatingLabelText="Longitude"
                                    id="longitude"
                                    name="longitude"
                                    onChange={[Function]}
                                    validatorListener={[Function]}
                                    validators={
                                                      Array [
                                                                        "required",
                                                                      ]
                                    }
                                    value={-81.234235}
                  />
                  <br />
                  <RaisedButton
                                    disabled={false}
                                    fullWidth={false}
                                    label="New Alarm"
                                    labelPosition="after"
                                    name="submitbutton"
                                    primary={false}
                                    secondary={false}
                                    type="submit"
                  />
</ValidatorForm>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Alarm: ",
                    "closed",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Alarm: ",
                  "closed",
                ],
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <TextValidator
                      errorMessages={
                                            Array [
                                                                  "this field is required",
                                                                ]
                      }
                      floatingLabelText="Latiutude"
                      id="latitude"
                      name="latitude"
                      onChange={[Function]}
                      validatorListener={[Function]}
                      validators={
                                            Array [
                                                                  "required",
                                                                ]
                      }
                      value={37.0212334}
/>,
                    <br />,
                    <TextValidator
                      errorMessages={
                                            Array [
                                                                  "this field is required",
                                                                ]
                      }
                      floatingLabelText="Longitude"
                      id="longitude"
                      name="longitude"
                      onChange={[Function]}
                      validatorListener={[Function]}
                      validators={
                                            Array [
                                                                  "required",
                                                                ]
                      }
                      value={-81.234235}
/>,
                    <br />,
                    <RaisedButton
                      disabled={false}
                      fullWidth={false}
                      label="New Alarm"
                      labelPosition="after"
                      name="submitbutton"
                      primary={false}
                      secondary={false}
                      type="submit"
/>,
                  ],
                  "name": "alarmform",
                  "onError": [Function],
                  "onSubmit": [Function],
                },
                "ref": "form",
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "errorMessages": Array [
                        "this field is required",
                      ],
                      "floatingLabelText": "Latiutude",
                      "id": "latitude",
                      "name": "latitude",
                      "onChange": [Function],
                      "validatorListener": [Function],
                      "validators": Array [
                        "required",
                      ],
                      "value": 37.0212334,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "br",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "errorMessages": Array [
                        "this field is required",
                      ],
                      "floatingLabelText": "Longitude",
                      "id": "longitude",
                      "name": "longitude",
                      "onChange": [Function],
                      "validatorListener": [Function],
                      "validators": Array [
                        "required",
                      ],
                      "value": -81.234235,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "br",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "disabled": false,
                      "fullWidth": false,
                      "label": "New Alarm",
                      "labelPosition": "after",
                      "name": "submitbutton",
                      "primary": false,
                      "secondary": false,
                      "type": "submit",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <RaisedButton
                disabled={false}
                fullWidth={false}
                label="Update View"
                labelPosition="after"
                onClick={[Function]}
                primary={false}
                secondary={false}
/>,
              "className": "common-paper-button-wrapper",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "disabled": false,
                "fullWidth": false,
                "label": "Update View",
                "labelPosition": "after",
                "onClick": [Function],
                "primary": false,
                "secondary": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
